src/fromager/overrides.py

```diff
@@ -108,12 +109,23 @@ def patches_for_requirement(
     override_name = pkgname_to_override_module(req.name)
     unversioned_patch_dir = patches_dir / override_name
     versioned_patch_dir = patches_dir / f"{override_name}-{version}"
+
+    unversioned_patch_files = list(unversioned_patch_dir.glob("*.patch"))
+    versioned_patch_files = list(versioned_patch_dir.glob("*.patch"))
+
+    # The list of files must exist to be joined to the global patch files
+    if variant:
+        unversioned_patch_files += list(
+            unversioned_patch_dir.glob(f"*.patch.{variant}")
+        )
+        versioned_patch_files += list(versioned_patch_dir.glob(f"*.patch.{variant}"))
+
     return itertools.chain(
         # Apply all of the unversioned patches first, in order based on
         # filename.
-        sorted(unversioned_patch_dir.glob("*.patch")),
+        sorted(unversioned_patch_files),
```

Comment:
I suggest to sort by the file's name, not the full path.

Reply:
Sounds complicated. I won't be able to recall the correct order in a week from now. How about we throw everything in a single list, then sort all files by their base name?

Reply:
We already give precedence to the files in the directory without the version. I think we don't have any of those downstream, so we probably wouldn't break anything making the change you propose. I guess it lets users manage precedence themselves using numerical prefixes, too, so maybe that's a good thing?

Reply:
Yes, that was the idea. The prefix is what decides the order. The user can easily insert a variant specific patch by using the same number as the global patch. `variant/0001-xx.patch` will have precedence over `0002-aa.patch`.